### PR TITLE
term: add documentation

### PR DESCRIPTION
**Pull request by snow-blade**
### Why
I noticed that there was no proper documentation for the term module, so i opened an issue and was tasked to do the task myself, so here it is. 
### What
I documented the term library and explained most of it's functions for a lambda user

### CODE REVIEW

File: vlib/term/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+// moves the cursor to the right
+term.cursor_forward()
+
+// moves the cursor to the left
+term.cursor_back()
+
+// show the cursor
+term.show_cursor()
+
+// hide the cursor
```
Comment (Delta456) [inline vlib/term/README.md:92]: hides

File: vlib/term/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// moves the cursor down
+term.cursor_down()
+
+// moves the cursor to the right
+term.cursor_forward()
+
+// moves the cursor to the left
+term.cursor_back()
+
+// show the cursor
```
Comment (Delta456) [inline vlib/term/README.md:89]: shows

File: vlib/term/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+// Transforms the given string into bold text
+term.bold(string)
+
+// Puts a strikethrough into the given string
+term.strikethrough(string)
+
+// Underlines the given string
+term.underline(string)
+
+// colors the background of the given string, replace <color> witha color of your choice
```
Comment (Delta456) [inline vlib/term/README.md:71]: with the*

File: vlib/term/README.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+This simple program covers many of the principal aspects of the `term ` module.
+
+# API
+
+Here are some functions you should be aware of in the `term `module:
+
+```v
+// returns the height and the width of the terminal
+width,height := term.get_terminal_size()
```
Comment (Delta456) [inline vlib/term/README.md:44]: could just do `term.get_terminal_size() (width, height)

File: vlib/term/README.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                        break
+                } else {
+                        var = os.input('press q to quit: ')
+                }
+        }
+}
+
+
+
+```
```
Comment (Delta456) [inline vlib/term/README.md:34]: Move this markdown ending to line 31

File: vlib/term/README.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+term.cursor_back()
+
+// show the cursor
+term.show_cursor()
+
+// hide the cursor
+term.hide_cursor
+
+
+```
```
Comment (Delta456) [inline vlib/term/README.md:96]: Moves this to line 94

**Comment by snow-blade** [comment]
Fixed the typos, thanks @Delta456

**Comment by medvednikov** [comment]
Thanks.

Documentation should be added to the code (`// Documentation` before each function). Then documentation page is generated automatically.

File: vlib/term/README.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+term.fail_message(string)
+
+// returns the string as yellow text to be printed on stdout 
+term.warning_message(string)
+
+//clears the entire terminal and leaves a blank one
+term.clear()
+
+// colors the output following the given color, replace <color> with a color of your choice
+term.<color>(string)
```
Comment (Delta456) [inline vlib/term/README.md:56]: Specify all the colors

**Comment by snow-blade** [comment]
@Delta456 , hmmm...what about now, how does it look?

**Comment by snow-blade** [comment]
I don't understand, why are those tests failing even if i did not change anything in the files affected by the tests ? the tests were all good and set before my latest commit.

File: vlib/term/README.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+You can use the `term` module to either color the output on a terminal or to decide on where to put the output in your terminal.
+
+For example let's make a simple program which prints colored text in the middle of the terminal.
+
+```v
+import term
+import os
+
+fn main() {
+        term.clear() // will clear everything and leave a blank terminal
```
Comment (Delta456) [inline vlib/term/README.md:16]: `// clears the content in the terminal`

File: vlib/term/README.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        width, height := term.get_terminal_size() // get the size of the terminal 
+        term.set_cursor_position(width / 2, height / 2) // Now we point the cursor to the middle of  the terminal 
+        println(term.strikethrough(term.bright_green("hello world")))  // Print green text
+        term.set_cursor_position(0, height) // Sets the position of the cursor to the bottom of the terminal
+        mut var := os.input('press q to quit: ')
+        // KEEP prompting the user until he enters the q key
```
Comment (Delta456) [inline vlib/term/README.md:22]: // Keep prompting until the user presses the `q` key

File: vlib/term/README.md (hunk 7 above)
Comment (Delta456) [reply vlib/term/README.md:56]: I think it should be like

```
// colors the output following in the given color; the available colors are: black, blue, yellow, green, cyan, gray
term.<color>(string)
```

File: vlib/term/README.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+// shows the cursor
+term.show_cursor()
+
+// hides the cursor
+term.hide_cursor()
+
+
+```
+
+
```
Comment (Delta456) [inline vlib/term/README.md:94]: Extra lines

File: vlib/term/README.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+// Puts a strikethrough into the given string
+term.strikethrough(string)
+
+// Underlines the given string
+term.underline(string)
+
+// colors the background of the given string, replace <color> with a color of your choice; the available colors are: black,blue,yellow,green,cyan,gray,bright_blue,bright_green,bright_red,bright_black,bright_cyan
+term.bg_<color>(string)
+
+// Sets the position of the cursor at a given place in the terminal
```
Comment (Delta456) [inline vlib/term/README.md:70]: `Sets` => `sets`

File: vlib/term/README.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+// returns the string as yellow text to be printed on stdout 
+term.warning_message(string)
+
+//clears the entire terminal and leaves a blank one
+term.clear()
+
+// colors the output following the given color, replace <color> with a color of your choice; the available colors are: black,blue,yellow,green,cyan,gray
+term.<color>(string)
+
+// Transforms the given string into bold text
```
Comment (Delta456) [inline vlib/term/README.md:58]: `Transforms` => `transforms`

File: vlib/term/README.md [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+term.bold(string)
+
+// Puts a strikethrough into the given string
+term.strikethrough(string)
+
+// Underlines the given string
```
Comment (Delta456) [inline vlib/term/README.md:64]: `Underline` => `underline`

File: vlib/term/README.md [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+//clears the entire terminal and leaves a blank one
+term.clear()
+
+// colors the output following the given color, replace <color> with a color of your choice; the available colors are: black,blue,yellow,green,cyan,gray
+term.<color>(string)
+
+// Transforms the given string into bold text
+term.bold(string)
+
+// Puts a strikethrough into the given string
```
Comment (Delta456) [inline vlib/term/README.md:61]: `Puts` => `puts`

File: vlib/term/README.md [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+term.cursor_back()
+
+// shows the cursor
+term.show_cursor()
+
+// hides the cursor
+term.hide_cursor()
+
+
+```
```
Comment (Delta456) [inline vlib/term/README.md:92]: This should be at line 90 and the lower extra lines has to be removed

File: vlib/term/README.md [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+fn main() {
+        term.clear() // will clear everything and leave a blank terminal
+        width, height := term.get_terminal_size() // get the size of the terminal 
+        term.set_cursor_position(width / 2, height / 2) // Now we point the cursor to the middle of  the terminal 
+        println(term.strikethrough(term.bright_green("hello world")))  // Print green text
+        term.set_cursor_position(0, height) // Sets the position of the cursor to the bottom of the terminal
+        mut var := os.input('press q to quit: ')
+        // KEEP prompting the user until he enters the q key
+        for {
+                if 'q' in var {
```
Comment (Delta456) [inline vlib/term/README.md:24]: I think it should just be `if var == 'q'`

**Comment by snow-blade** [comment]
why is the test still failing?

File: vlib/term/README.md (hunk 10 above)
Comment (Delta456) [reply vlib/term/README.md:94]: Remove these

File: vlib/term/README.md (hunk 15 above)
Comment (Delta456) [reply vlib/term/README.md:92]: Please remove

**Comment by medvednikov** [comment]
Master branch can be broken sometimes. It's now green. It's unrelated.

File: vlib/term/README.md [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+fn main() {
+        term.clear() // will clear everything and leave a blank terminal
+        width, height := term.get_terminal_size() // get the size of the terminal 
+        term.set_cursor_position(width / 2, height / 2) // Now we point the cursor to the middle of  the terminal 
+        println(term.strikethrough(term.bright_green("hello world")))  // Print green text
+        term.set_cursor_position(0, height) // Sets the position of the cursor to the bottom of the terminal
+        mut var := os.input('press q to quit: ')
+        // KEEP prompting the user until he enters the q key
+        for {
+                if  var=='q' {
```
Comment (Delta456) [inline vlib/term/README.md:24]: `if var == 'q'`

File: vlib/term/README.md (hunk 8 above)
Comment (Delta456) [reply vlib/term/README.md:16]: Please do this

File: vlib/term/README.md (hunk 9 above)
Comment (Delta456) [reply vlib/term/README.md:22]: Please do this as well

File: vlib/term/README.md [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+term.cursor_back()
+
+// shows the cursor
+term.show_cursor()
+
+// hides the cursor
+term.hide_cursor()
+```
+
+
```
Comment (Delta456) [inline vlib/term/README.md:92]: Extra line

**Comment by snow-blade** [comment]
hey @Delta456 can you please suggest all the changes at once, it's a bit not cool when you suggest changes one by one each time i commit

**Pull request merged**